icu4c/source/test/intltest/ustrtest.cpp

```diff
@@ -2310,3 +2311,21 @@ void UnicodeStringTest::TestUnicodeStringInsertAppendToSelf() {
     str.insert(2, sub);
     assertEquals("", u"abbcdcde", str);
 }
+
+void UnicodeStringTest::TestLargeAppend() {
+    IcuTestErrorCode status(*this, "TestLargeAppend");
+    // Make a large UnicodeString
+    int32_t len = 0xAFFFFFF;
+    UnicodeString str;
+    char16_t *buf = str.getBuffer(len);
+    for (int32_t i = 0; i < len; i++) {
+      *buf++ = u'a';
+    }
+    str.releaseBuffer(len);
+    UnicodeString dest;
+    // Append it 16 times
+    for (int32_t i = 0; i < 16; i++) {
```

Comment:
Note: 0xAFFFFFF times 16 is 0xA4FFFFF1, which is greater than INT32_MAX, which is 0x7FFFFFFF.  Can you add that as a comment?

Reply:
done